end_to_end/gemma/7b/2_test_gemma.sh

```diff
@@ -10,17 +10,27 @@
 # 4. Ahead of Time Compilation for running Gemma 7B on v5e-256
 
 set -ex
-idx=$(date +%Y-%m-%d-%H-%M)
 export MODEL_VARIATION='7b'
 
-# Non-Googlers please remember to MODEL_BUCKET to GCS bucket where this script uses internal buckets for testing.
-export MODEL_BUCKET=gs://maxtext-gemma
+if [ -z "${MODEL_BUCKET}" ]; then
+    # Non-Googlers please remember to MODEL_BUCKET to GCS bucket where this script uses internal buckets for testing.
+    export MODEL_BUCKET=gs://maxtext-gemma
+    idx=$(date +%Y-%m-%d-%H-%M)
+    
+else
+    idx=$(awk -F'/' '{print $NF}' <<<"${MODEL_BUCKET}")
+
+fi
+
+echo "Converted checkpoint is stored at ${MODEL_BUCKET} and idx = ${idx}"
+
+
 # Non-Googlers please remember to point `DATASET_PATH` to the GCS bucket where you have your training data
 export DATASET_PATH=gs://maxtext-dataset
 # Non-Googlers please remember to point `BASE_OUTPUT_DIRECTORY` to a GCS bucket that you own, this bucket will store all the files generated by MaxText during a run
 export BASE_OUTPUT_DIRECTORY=gs://runner-maxtext-logs
 # We define `CONVERTED_CHECKPOINT` to refer to the checkpoint subdirectory. This way it is easier to use this path in the `train.py` and `decode.py` commands
-export CONVERTED_CHECKPOINT=${MODEL_BUCKET}/${MODEL_VARIATION}/${idx}/0/items
+export CONVERTED_CHECKPOINT=${MODEL_BUCKET}/${MODEL_VARIATION}/0/items
```

Comment:
Just confirming, `idx` has been absorbed into `MODEL_BUCKET`, so this will still be distinct for each run?

Reply:
Right, since `MODEL_BUCKET` has the `idx` absorbed into it, we are getting `CONVERTED_CHECKPOINT` distinct in each run. We are also pulling out `idx` in line 21 from `MODEL_BUCKET`, and used in the different `RUN_NAME` vars in line 24, 47, 56, these `RUN_NAME` vars would make the outputs of `end_to_end/gemma/7b/2_test_gemma.sh` distinct for each run.

Is this what you meant?